draft-ietf-httpbis-digest-headers.md

```diff
@@ -501,6 +500,47 @@ to collision attacks [CMU-836068].
 The SHA algorithm is NOT RECOMMENDED as it's now vulnerable
 to collision attacks [IACR-2019-459].
 
+# Use of Digest when acting on resources
+
+As the representation-data-digest is calculated on the selected representation,
+you need all the representation-metadata to compute it.
+
+Methods acting on resources, like PATCH and POST
+while seamingly may convey partial representations
+are semantically acting on resources.
+
+PATCH representation metadata apply to the patch document
+and not to the target resource, as stated in Section 2 of {{?RFC5789}},
+and should instead use a proper content-type, eg {{?RFC7396}}.
+
+In these cases, the representation-data-digest
+MUST be computed on the action of the payload.
+
+The response instead can convey the representation-data-digest
+of the representation as long as it contains all the required
+representation metadata.
```

Comment:
Alternative suggestion for above 4 pargraphs:

Methods such as PATCH or POST may appear to convey partial representations but are semantically acting on resources. In these cases the representation-data-digest MUST be computed on the action of the payload. The response to such requests can convey the representation-data-digest.

In the case of PATCH a request conveys representation metadata that applies to the patch document and not the target resource (see ,Section 2 of {{?RFC5789}}). 

The following example shows a PATCH request (using a valid content-type defined in {{?RFC7396}} and a response that contains the representation-data-digest of the patched representation.